static/js/reactions.js

```diff
@@ -109,6 +115,51 @@ $(document).on('click', '.reaction-popover-reaction', function () {
 $(document).on('input', '.reaction-popover-filter', filter_emojis);
 $(document).on('keydown', '.reaction-popover-filter', maybe_select_emoji);
 
+exports.reaction_navigate = function (e, event_name) {
+    var first_emoji = get_emoji_from_index(0);
+    var selected_emoji = $(".reaction-popover-reaction").filter(":focus")[0];
+    var reaction_show_list = $(".reaction-popover-reaction").filter(function () {
+        return this.style.display === "block" || this.style.display === "";
+    }).toArray();
+    var selected_index = reaction_show_list.findIndex(function (emoji) {
+        return emoji === selected_emoji;
+    });
+    var next_index;
+    if (event_name === 'down_arrow') {
+        if ($('.reaction-popover-filter').is(':focus') && first_emoji) { // move down into emoji map
+            $(first_emoji).focus();
+        } else if (selected_emoji) {
+            next_index = selected_index + 6; // 6 is the length of the reaction box
+            if (next_index < reaction_show_list.length) {
+                $(reaction_show_list[next_index]).focus();
+            }
+        }
+        e.preventDefault();
+    } else if (event_name === 'up_arrow') {
+        if (selected_emoji && selected_index < 6) { // move up into reaction filter
+            $('.reaction-popover-filter').focus();
+        } else if (selected_emoji) {
+            next_index = selected_index - 6; // 6 is the length of the reaction box
+            if (next_index < reaction_show_list.length) {
+                $(reaction_show_list[next_index]).focus();
+            }
+        }
+        e.preventDefault();
+    } else if (event_name === 'left_arrow' && selected_emoji) {
```

Comment:
Is there a way to handle `selected_emoji` more toward the top of the function?  It seems like if we don't have an emoji selected, we should just return early, correct?